packages/core/http/core-http-server/src/versioning/types.ts

```diff
@@ -102,16 +67,15 @@ export interface VersionedRouter<Ctx extends RqCtx = RqCtx> {
   patch: VersionedRouteRegistrar<'patch', Ctx>;
   /** @experimental */
   delete: VersionedRouteRegistrar<'delete', Ctx>;
-  /** @experimental */
-  options: VersionedRouteRegistrar<'options', Ctx>;
 }
 
 /** @experimental */
 export type RequestValidation<P, Q, B> = RouteValidatorFullConfig<P, Q, B>;
 
 /** @experimental */
 export interface ResponseValidation<R> {
-  body: RouteValidationFunction<R> | Type<R>;
+  [statusCode: number]: { body: RouteValidationFunction<R> | Type<R> };
```

Comment:
Following OAS's example here. For the responses we added validation for specific status codes. At the moment this is used to scope validation to a specific status code, but we could go further and throw if an unknown status code was provided.

Note: output validation is intended to run in dev only.